File: ejercicios/python/ejercicio079.py
numero = int(input("Ingrese un numero entero: "))
base = int(input("Ingrese la base del numero: "))

base2 = []
base8 = []
base10 = []

if numero >= 0:
    for i in range(0,2):
        base2.append(i)
    
    for i in range(0,8):
        base8.append(i)
    
    for i in range(0,10):
        base10.append(i)
    
    n = numero
    
    if base == 2:
        while (n > 0):
            digito = n % 10
            estaen = False
            if digito in base2:
                estaen = True
            
            n = n // 10
            
            if  not estaen: 
                n = -1
        
        if n == 0: 
            print(str(numero) + " esta en Base 2")
        else:
            print(str(numero) + " No esta en Base 2")
    elif base == 8:
        while (n > 0):
            digito = n % 10
            estaen = False
            if digito in base8:
                estaen = True
            n = n // 10
            
            if not estaen: 
                n = -1
        
        if n == 0: 
            print(str(numero) + " esta en Base 8")
        else:
            print(str(numero) + " No esta en Base 8")
    elif base == 10:
        while (n > 0):
            digito = n % 10
            estaen = False
            if digito in base10:
                estaen = True
            n = n // 10
            
            if not estaen: 
                n = -1
        
        if n == 0: 
            print(str(numero) + " esta en Base 10")
        else:
            print(str(numero) + " No esta en Base 10")      
    else:
        print("La base " + str(base) + " no esta definida")
else:
    print("Error en el ingreso de datos, " + str(numero) + " debe ser >= 0")

File: ejercicios/c++/ejercicio079.cpp
#include<iostream>
using namespace std;

int main() {

	int base;
	int base10[10];
	int base2[2];
	int base8[8];
	int digito;
	bool estaen;
	int n;
	int numero;

	cout << "Ingrese un numero entero: ";
	cin >> numero;
	cout << "Ingrese la base del numero: ";
	cin >> base;
	
	if (numero >= 0) {
		for (int i = 0; i <= 1; i++) {
			base2[i] = i;
		}
		
		for (int i = 0; i <= 7; i++) {
			base8[i] = i;
		}
		
		for ( int i = 0; i <= 9; i++) {
			base10[i] = i;
		}
		
		n = numero;
		
		switch ( base ) {
		case 2:
			while ( n > 0) {
				digito = n % 10;
				estaen = false;
				for (int i = 0; i <= 1; i++) {
					if (base2[i] == digito) {
						estaen = true;
						i = 2;
					}
				}
				n = n / 10 ;
				if ( !estaen ) {
					n = -1;
				}
			}
			if (n == 0) {
				cout << numero << " esta en Base 2" << endl;
			} else {
				cout << numero << " No esta en Base 2" << endl;
			}
			break;
		case 8:
			while ( n>0 ) {
				digito = n % 10;
				estaen = false;
				for (int i = 0; i <= 7; i++) {
					if (base8[i] == digito) {
						estaen = true;
						i = 8;
					}
				}
				n = n / 10;
				if ( !estaen ) {
					n = -1;
				}
			}
			if (n == 0) {
				cout << numero << " esta en Base 8" << endl;
			} else {
				cout << numero << " No esta en Base 8" << endl;
			}
			break;
		case 10:
			while (n > 0) {
				digito = n % 10;
				estaen = false;
				for (int i = 0; i <= 9; i++) {
					if (base10[i] == digito) {
						estaen = true;
						i = 10;
					}
				}
				n = n/10 ;
				if ( !estaen ) {
					n =- 1;
				}
			}
			if (n == 0) {
				cout << numero << " esta en Base 10" << endl;
			} else {
				cout << numero << " No esta en Base 10" << endl;
			}
			break;
		default:
			cout << "La base " << base << " no esta definida" << endl;
			break;
		}
	} else {
		cout << "Error en el ingreso de datos, " << numero << " debe ser >= 0" << endl;
	}

	return 0;

}

